source/SIL.AppBuilder.Portal.Frontend/src/sockets/data-socket.tsx

```diff
@@ -98,9 +101,12 @@ export default function LiveDataManager({ children }) {
     [isConnected, isTesting]
   );
 
+  console.log('socket:', isConnected);
   return children({ socket: dataClient, pushData, subscriptions, isConnected, connectionState });
 }
 
+export default memo(LiveDataManager);
```

Comment:
a theory about why the socket connection was getting lost was that maybe the the component is getting over-re-rendered, and we were losing the individual property-based memoization somehow -- that turned out to not be the case, so useCallback was added to creating the hub Factory and the client itself. that seemed to improve stability a lot - but has not yet fixed the overall problem where we try to send messages to a disconnected client.